vent/sqldb/adapters/sqlite_adapter.go

```diff
@@ -0,0 +1,309 @@
+		query += updValues
+	} else {
+		query += fmt.Sprintf("ON CONFLICT ON CONSTRAINT %s_pkey DO NOTHING", table.Name)
+	}
+	query += ";"
+
+	upsertQuery.Query = query
+	return upsertQuery
+}
+
+// LastBlockIDQuery returns a query for last inserted blockId in log table
+func (adapter *SQLiteAdapter) LastBlockIDQuery() string {
+	query := `
+		WITH ll AS (
+			SELECT MAX(%s) AS %s FROM %s WHERE %s = $1 
+		)
+		SELECT COALESCE(%s, '0') AS %s 
+			FROM ll LEFT OUTER JOIN %s log ON (ll.%s = log.%s);`
+
+	return fmt.Sprintf(query,
+		types.SQLColumnNameId,          //max
+		types.SQLColumnNameId,          //as
+		types.SQLLogTableName,          //from
+		types.SQLColumnNameEventFilter, //where
+
+		types.SQLColumnNameHeight,                    //coalesce
+		types.SQLColumnNameHeight,                    //as
+		types.SQLLogTableName,                        //from
+		types.SQLColumnNameId, types.SQLColumnNameId) //on
+
+}
+
+// FindTableQuery returns a query that checks if a table exists
+func (adapter *SQLiteAdapter) FindTableQuery() string {
+	query := "SELECT COUNT(*) found FROM %s WHERE %s = $1;"
+
+	return fmt.Sprintf(query,
+		types.SQLDictionaryTableName, //from
+		types.SQLColumnNameTableName) //where
+
+}
+
+// TableDefinitionQuery returns a query with table structure
+func (adapter *SQLiteAdapter) TableDefinitionQuery() string {
+	query := `
+		SELECT 
+			%s,%s,%s,%s 
+		FROM 
+			%s 
+		WHERE 
+			%s=$1 
+		ORDER BY
+			%s;`
+
+	return fmt.Sprintf(query,
+		types.SQLColumnNameColumnName, types.SQLColumnNameColumnType, //select
+		types.SQLColumnNameColumnLength, types.SQLColumnNamePrimaryKey, //select
+		types.SQLDictionaryTableName,   //from
+		types.SQLColumnNameTableName,   //where
+		types.SQLColumnNameColumnOrder) //order by
+
+}
+
+// AlterColumnQuery returns a query for adding a new column to a table
+func (adapter *SQLiteAdapter) AlterColumnQuery(tableName string, columnName string, sqlColumnType types.SQLColumnType, length int, order int) (string, string) {
+	sqlType, _ := adapter.TypeMapping(sqlColumnType)
+	if length > 0 {
+		sqlType = fmt.Sprintf("%s(%d)", sqlType, length)
+	}
+
+	query := fmt.Sprintf("ALTER TABLE %s ADD COLUMN %s %s;",
+		tableName,
+		columnName,
+		sqlType)
+
+	dictionaryQuery := fmt.Sprintf(
+		`INSERT INTO %s 
+					(%s,%s,%s,%s,%s,%s) 
+				VALUES 
+					('%s','%s',%d,%d,%d,%d);`,
+
+		types.SQLDictionaryTableName,
+
+		types.SQLColumnNameTableName, types.SQLColumnNameColumnName,
+		types.SQLColumnNameColumnType, types.SQLColumnNameColumnLength,
+		types.SQLColumnNamePrimaryKey, types.SQLColumnNameColumnOrder,
+
+		tableName, columnName,
+		sqlColumnType, length,
+		0, order)
+
+	return query, dictionaryQuery
+}
+
+// SelectRowQuery returns a query for selecting row values
+func (adapter *SQLiteAdapter) SelectRowQuery(tableName string, fields string, indexValue string) string {
+	return fmt.Sprintf("SELECT %s FROM %s WHERE _height='%s';", fields, tableName, indexValue)
+}
```

Comment:
`fmt.Sprintf("SELECT %s FROM %s WHERE _height='%s';", fields, tableName, indexValue)` should be
`fmt.Sprintf("SELECT %s FROM %s WHERE %s='%s';", fields, tableName, types.SQLColumnNameHeight, indexValue)` to be consistent with other similar queries

Reply:
ready